code/renderers/vue3/src/decorateStory.ts

```diff
@@ -64,8 +81,31 @@ export function decorateStory(
         return story;
       }
 
-      return prepare(decoratedStory, h(story, context.args)) as VueRenderer['storyResult'];
+      const innerStory = () => (story ? h(story, context.args) : null);
+      return prepare(decoratedStory, innerStory) as VueRenderer['storyResult'];
     },
     (context) => prepare(storyFn(context)) as LegacyStoryFn<VueRenderer>
   );
 }
+/**
+ * update the context with the update object from the decorator in reactive way
+ * @param context
+ * @param update
+ */
+export function updateReactiveContext(
+  context: StoryContext<VueRenderer>,
+  update: StoryContextUpdate<Partial<Args>> | undefined
+) {
+  if (update) {
+    const { args, argTypes } = update;
+    if (args && !argTypes) {
+      const deepCopy = JSON.parse(JSON.stringify(args)); // avoid reference to args
```

Comment:
What if the args are not primitives? Function vs classes?
Do we need the deep copy?

Reply:
i think i will remove or change with 
```t
const argCopy = { ...args }
```
i have to test it again, the issue is if the updated args has reference to context.args which is most the time , in this case you will applying change on both of them, what i need is the values only.